names.Rmd

```diff
@@ -174,57 +257,58 @@ There are many ways to fix a non-syntactic name. Here's how our logic compares t
  * Same: Definition of what is syntactically valid.
    - Claim: If `syn_name` is a name that we have made syntactic, then
      `syn_name == make.names(syn_name)`. If you find a counterexample, tell us!
- * Same: Invalid characters are replaced with `.`.
- * Different: We always fix a name by prepending a `.`. [base::make.names()](https://stat.ethz.ch/R-manual/R-patched/library/base/html/make.names.html)
-   sometimes prefixes with `X` and at other times appends a `.`.
+ * Same: An invalid character is replaced with a dot `.`.
+ * Different: We always fix a name by prepending a dot `.`. [base::make.names()](https://stat.ethz.ch/R-manual/R-patched/library/base/html/make.names.html)
+   sometimes prefixes with `X` and at other times appends a dot `.`.
  * Different: We treat `NA` and `""` the same: both become `.`. This is because
    we first make names minimal. [base::make.names()](https://stat.ethz.ch/R-manual/R-patched/library/base/html/make.names.html) turns `NA` into `"NA."` and `""` into `"X"`.
- * Different: We turn `...` into `....` and `..i` into `...i` (`i` is a
-   number). [base::make.names()](https://stat.ethz.ch/R-manual/R-patched/library/base/html/make.names.html) does not modify `...` or `..i`, which could
+ * Different: We turn `...` into `....` and `..j` into `...j`, where `j` is a
```

Comment:
```suggestion
 * Different: We turn `...` and `..j` into `...j`, where `j` is a
```

Reply:
Here, in `make_syntactic()`, we actually do turn `...` into `....`. That never happens with our name repair functions, because the `...` is intercepted earlier inside `unique_names()` and treated like `""`. I too fell for this and made this edit, then reversed it.

Reply:
I agree it's a little icky and confusing and I thought about this specifically.

The alternative is to turn `...` into `.`. But then that violates our over approach to name repair, which is to keep adding dots to the front until it's syntactic.

There is no perfect solution for `...`. Luckily it's rarely proposed as a variable name.

Reply:
Got you. Maybe we can treat this as a "prepending a dot" exercise too, and combine with the "Different" bullet point at https://github.com/tidyverse/principles/pull/51/files#diff-19bde92556fa22757ad23b29d8a4002aR261?

Reply:
Yeah, nice. I've done that.